lib/iex/test/iex/autocomplete_test.exs

```diff
@@ -162,4 +162,36 @@ defmodule IEx.AutocompleteTest do
     assert expand('EList') == {:yes, '.', []}
     assert expand('EList.map') == {:yes, [], ['map/2', 'mapfoldl/3', 'mapfoldr/3']}
   end
+
+  test "completion for functions added when a module is reloaded" do
+    filename = "sample.ex"
+    content = """
+    defmodule Sample do
+    def foo(), do: 0
+    end
+    """
+
+    reloaded_content = """
+    defmodule Sample do
+    def foo(), do: 0
+    def foobar(), do: 0
+    end
+    """
+    File.write!(filename, content)
+    compiler_options = :elixir_config.get(:compiler_options)
```

Comment:
Could we not fiddle with compiler options? Can't we just pick another module name? Like Autocomplete?